test/server/auth.spec.js

```diff
@@ -1,165 +1,192 @@
+        return res.end();
+      }
+      let body = '';
+      req.on('data', (chunk) => {
+        body += chunk;
+      });
+      let requrl = url.parse(req.url);
+      let ok;
+      req.on('end', () => {
+        let reqdata = JSON.parse(body);
+        if (requrl.pathname === constants.AUTH_URL_TOKEN) {
+          if (reqdata.token === 'abc' && isSubset(reqdata.groups, ['1', '2', '3'])) {
+            ok = true;
+          } else {
+            ok = false;
+          }
+          res.end(JSON.stringify({ok: ok}));
+        } else if (requrl.pathname === constants.AUTH_URL_USER) {
+          if (reqdata.user === 'alice' && isSubset(reqdata.groups, ['1', '2', '3'])) {
+            ok = true;
+          } else {
+            ok = false;
+          }
+          res.end(JSON.stringify({ok: ok}));
+        } else {
+          res.statusCode = 404;
+          res.end();
+        }
+      });
+    });
+    dummyAuthServ.listen(9898);
+    closeServ = dummyAuthServ.close;
   });
 
   it('Input validation', function() {
     expect( () => {new DataAccess();} ).toThrowError(ReferenceError,
-      'Database handle is required');
+      'Authorisation type is required');
+    expect( () => {new DataAccess('badauth');} ).toThrowError(ReferenceError,
+      'Unknown authorisation type');
+    expect( () => {new DataAccess(constants.AUTH_TYPE_USER);} ).not.toThrow();
+    expect( () => {new DataAccess(constants.AUTH_TYPE_TOKEN);} ).not.toThrow();
     expect( () => {
-      let da = new DataAccess({});
+      let da = new DataAccess(constants.AUTH_TYPE_USER);
       da.authorise();
-    }).toThrowError(ReferenceError, 'Username is required');
+    }).toThrowError(ReferenceError, 'Identifier is required');
     expect( () => {
-      let da=new DataAccess({});
+      let da = new DataAccess(constants.AUTH_TYPE_TOKEN);
+      da.authorise();
+    }).toThrowError(ReferenceError, 'Identifier is required');
+    expect( () => {
+      let da = new DataAccess(constants.AUTH_TYPE_USER);
       da.authorise('alice');
     }).toThrowError(Error, 'Access groups array is not available');
     expect( () => {
-      let da=new DataAccess({});
+      let da = new DataAccess(constants.AUTH_TYPE_USER);
       da.authorise('alice', 2);
     }).toThrowError(Error, 'Access groups array is not available');
     expect( () => {
-      let da=new DataAccess({});
+      let da = new DataAccess(constants.AUTH_TYPE_USER);
       da.authorise('alice', []);
     }).toThrowError(Error, 'Access groups array is not available');
   });
 
   it('Authorisation failed - username is all whitespace', function(done) {
-    var da = new DataAccess({});
-    da.on('failed', (username, reason) => {
-      expect(username).toBe('   ');
-      expect(reason).toBe('Invalid user "   "');
+    var da = new DataAccess(constants.AUTH_TYPE_USER);
+    da.on('failed', (reason) => {
+      expect(reason).toBe('Failed to get authorisation info: Error: Invalid identifier "   "');
```

Comment:
probably better if we use
.toMatch(/Error: Invalid indetifier/i)
or something similar?